pkg/test/istio.io/examples/example.go

```diff
@@ -14,31 +14,35 @@
 package examples
 
 import (
-	"fmt"
 	"os"
+	"path"
 	"testing"
 
+	"istio.io/istio/pkg/test/env"
 	"istio.io/istio/pkg/test/framework/components/environment/kube"
 	"istio.io/istio/pkg/test/scopes"
 
 	"istio.io/istio/pkg/test/framework"
 )
 
-type outputType string
+type outputType outputTypeImpl
 
-const (
+var (
 	// TextOutput describes a test which returns text output
-	TextOutput outputType = "text"
+	TextOutput outputType = textOutput{}
 	// YamlOutput describes a test which returns yaml output
-	YamlOutput outputType = "yaml"
+	YamlOutput outputType = yamlOutput{}
 	// JSONOutput describes a test which returns json output
-	JSONOutput outputType = "json"
+	JSONOutput outputType = jsonOutput{}
```

Comment:
It's bit odd to have publicly exposed variables of a publicly exposed type. What does that mean to someone importing the package?

Reply:
Thanks. Hiding the implementation of the jsonOutput, yamlOutput, and textOutput behind a single instance in the framework such that a user can do:

```golang
RunScript("curl-foo-bar-legacy.sh", examples.TextOutput, examples.GetCurlVerifier([]string{"200", "200", "200"})).
```

Do you think it would be better just to make the output type public and expose it to the user as: 
```golang
RunScript("curl-foo-bar-legacy.sh", examples.TextOutput{}, examples.GetCurlVerifier([]string{"200", "200", "200"})).
```

Also, I created the outputType interface to allow us to customize the output for JSON/YAML/Text. Not sure what sort of customizations we're expecting though. Are we expecting just the file's extension to change for each output type? Or are we expecting additional functionality such as pretty printing?

Reply:
I like the simpler examples.TextOutput form, I'm just surprised that you're allowed to expose a variable publicly that's of a private type. But if it works...

In terms of output customization, it's mostly about the file extension, which the site uses to colorize the content. It's conceivable that in certain cases we might want pretty printing, but we can wait until demand materializes. Your setup would enable us to support that easily.